[HIP] Compilation errors with non-blocking stream creation

```cpp
occa/src/occa/internal/modes/hip/device.cpp: In member function ‘virtual occa::modeStream_t* occa::hip::device::createStream(const occa::json&)’:
occa/src/occa/internal/modes/hip/device.cpp:127:24: error: ‘hipStreamCreateWithFlag’ was not declared in this scope
                        hipStreamCreateWithFlag(&hipStream, hipStreamNonBlocking));
                        ^~~~~~~~~~~~~~~~~~~~~~~

```

--- a/src/occa/internal/modes/hip/buffer.cpp
+++ b/src/occa/internal/modes/hip/buffer.cpp
@@ -18,7 +18,8 @@ namespace occa {
           OCCA_HIP_ERROR("Device: hostFree()",
                            hipHostFree(ptr));
         } else if (hipPtr) {
-          hipFree((void*) hipPtr);
+          OCCA_HIP_ERROR("Device: free()",
+			 hipFree((void*) hipPtr));
         }
       }
       ptr = nullptr;

--- a/src/occa/internal/modes/hip/device.cpp
+++ b/src/occa/internal/modes/hip/device.cpp
@@ -124,7 +124,7 @@ namespace occa {
                      hipSetDevice(deviceID));
       if (props.get<bool>("nonblocking", false)) {
         OCCA_HIP_ERROR("Device: createStream - NonBlocking",
-                       hipStreamCreateWithFlag(&hipStream, hipStreamNonBlocking));
+                       hipStreamCreateWithFlags(&hipStream, hipStreamNonBlocking));
       } else {
         OCCA_HIP_ERROR("Device: createStream",
                        hipStreamCreate(&hipStream));

--- a/src/occa/internal/modes/hip/polyfill.hpp
+++ b/src/occa/internal/modes/hip/polyfill.hpp
@@ -275,7 +275,7 @@ namespace occa {
     return OCCA_HIP_IS_NOT_ENABLED;
   }
 
-  inline hipError_t hipStreamCreateWithFlag(hipStream_t *phStream, unsigned int flags) {
+  inline hipError_t hipStreamCreateWithFlags(hipStream_t *phStream, unsigned int flags) {
     return OCCA_HIP_IS_NOT_ENABLED;
   }
 

--- a/src/occa/internal/modes/hip/utils.cpp
+++ b/src/occa/internal/modes/hip/utils.cpp
@@ -19,7 +19,7 @@ namespace occa {
 
     int getDeviceCount() {
       int deviceCount = 0;
-      hipGetDeviceCount(&deviceCount);
+      static_cast<void>(hipGetDeviceCount(&deviceCount));
       return deviceCount;
     }
 

